lib/resty/jwt.lua

```diff
@@ -285,36 +310,35 @@ end
 -- @function parse token - this can be JWE or JWT token
 -- @param token string
 -- @return jwt/jwe tables
-local function parse(secret, token_str)
+local function parse(self, secret, token_str)
   local tokens = split_string(token_str, str_const.regex_split_dot)
   local num_tokens = #tokens
   if num_tokens == 3 then
     return  parse_jwt(tokens[1], tokens[2], tokens[3])
   elseif num_tokens == 4  then
-    return parse_jwe(secret, tokens[1], "", tokens[2], tokens[3],  tokens[4])
+    return parse_jwe(self, secret, tokens[1], nil, tokens[2], tokens[3],  tokens[4])
   elseif num_tokens == 5 then
-    return parse_jwe(secret, tokens[1], tokens[2], tokens[3],  tokens[4], tokens[5])
+    return parse_jwe(self, secret, tokens[1], tokens[2], tokens[3],  tokens[4], tokens[5])
   else
     error({reason=str_const.invalid_jwt})
   end
 end
 
-
 --@function jwt encode : it converts into base64 encoded string. if input is a table, it convets into
 -- json before converting to base64 string
 --@param payloaf
 --@return base64 encoded payloaf
-function _M.jwt_encode(self, ori)
+function _M.jwt_encode(self, ori, is_payload)
```

Comment:
not sure how I feel about this one

Reply:
I added this so that actual payload can have diff format than json (like msgpack ) and developer can provide decoder for that. Since this function is also used for decoding header. I am passing flag to determine that. Default behavior  is not changed so it is 100% backward compatible